parquet-column/src/main/java/org/apache/parquet/column/ParquetProperties.java

```diff
@@ -74,19 +78,28 @@ public static WriterVersion fromString(String name) {
   private final int dictionaryPageSizeThreshold;
   private final WriterVersion writerVersion;
   private final boolean enableDictionary;
+  // TODO - Should this even hold an allocator? this isn't really a property
+  private ByteBufferAllocator allocator;
```

Comment:
I'm not sure what's the best thing to do here. This is more of a configuration object.
this object is immutableplease make all fields final.